From 4e5d302cdff64188c7fc6a0ad3c1c9553a3301d7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pavel=20B=C5=99ezina?= <pbrezina@redhat.com>
Date: Fri, 20 Apr 2012 14:32:01 +0200
Subject: [PATCH 2/2] Talloc doc: talloc_strdup_append does not return
 duplicated string

---
 lib/talloc/talloc.h |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/lib/talloc/talloc.h b/lib/talloc/talloc.h
index 3e2e9ec..05e6292 100644
--- a/lib/talloc/talloc.h
+++ b/lib/talloc/talloc.h
@@ -1317,7 +1317,7 @@ char *talloc_strdup(const void *t, const char *p);
  *
  * @param[in]  a        The string you want to append.
  *
- * @return              The duplicated string, NULL on error.
+ * @return              The concatenated strings, NULL on error.
  *
  * @see talloc_strdup()
  * @see talloc_strdup_append_buffer()
@@ -1352,7 +1352,7 @@ char *talloc_strdup_append(char *s, const char *a);
  *
  * @param[in]  a        The string you want to append.
  *
- * @return              The duplicated string, NULL on error.
+ * @return              The concatenated strings, NULL on error.
  *
  * @see talloc_strdup()
  * @see talloc_strdup_append()
@@ -1404,7 +1404,7 @@ char *talloc_strndup(const void *t, const char *p, size_t n);
  * @param[in]  n        The number of characters you want to append from the
  *                      string.
  *
- * @return              The duplicated string, NULL on error.
+ * @return              The concatenated strings, NULL on error.
  *
  * @see talloc_strndup()
  * @see talloc_strndup_append_buffer()
@@ -1442,7 +1442,7 @@ char *talloc_strndup_append(char *s, const char *a, size_t n);
  * @param[in]  n        The number of characters you want to append from the
  *                      string.
  *
- * @return              The duplicated string, NULL on error.
+ * @return              The concatenated strings, NULL on error.
  *
  * @see talloc_strndup()
  * @see talloc_strndup_append()
-- 
1.7.6.5

